Support for Sigil II

dgondos opened the pull request:
# Description

Adding support for Sigil II as well as build instructions/helpers for Linux/Dosbox

![image](https://github.com/Sakitoshi/mbf_sigil/assets/11691500/00ba0b30-624b-4675-af94-3d2b1d03ed4f)

# Changes

- Tried to find all places where there's an assumption on episode size and extended them to support 6 episodes
- Sigil II specific hardcoded map names and end text
- Sigil II specific intermission screen
- Build instructions and helpers for Linux/Dosbox
- I turned `modifiedgame` into an integer rather than a boolean to count the amount of PWADs requested by the user. It's used later to handle the restriction on max number of episodes. This still isn't great because we don't do any checks on whether those PWADs are Sigil & Sigil II - so it basically means that we assume that if you load two PWADs then one will be EP5 and the other EP6. Previously we assumed that any new PWAD will be EP5. If I understand correctly the way they solved this in prboom is they just set the max number of episodes to a high number (7) and they no longer check if the user requests an episode which is between 4 and 7 but doesn't map to a real episode in a loaded wad.

# Known issues

- Sigil II can only be loaded if Sigil I is also loaded. If you load just Sigil II alone, it doesn't show up in the episode selection screen. I believe this is because of implicit assumptions in MBF's code that there are no "episode gaps". This is not fixed with this patch. This might mean that when you play Sigil I with Sigil I and II both loaded, stuff won't work as expected if some filenames were re-used inside the WAD but the content changed.
- The title screen music's length is not taken into account (cuts off early before the whole song plays).

# Testing

- Tested the title screen, E6M1 first room and E5M1 first room (for regression). All worked well in dosbox.
- Tested the same way on 86Box as well (with Win98's DOS). No issues.
- I'll play through the wad as well on real hardware. Not yet started though.

# Binary

[mbf_snm10_sigil2_2.zip](https://github.com/Sakitoshi/mbf_sigil/files/13727999/mbf_snm10_sigil2_2.zip)

(note: copy this .exe over the standard MBF_SMN release from [here](https://github.com/Sakitoshi/mbf_sigil/releases/tag/v2.04-snm10))

dgondos committed:
build helpers and instructions for linux/dosbox

dgondos committed:
error handling for dosbox

dgondos committed:
basic sigil II support

dgondos committed:
specific differences between sigil I & II

dgondos committed:
add screenshot

dgondos committed:
readme: adding confirmed djgpp version which works

dgondos committed:
dpmi needed for build as well

Sakitoshi commented:
very nice. compiled it myself and it works without issues on my end.
even tested on real hardware (IBM ThinkPad 365XD) and it works fine, well, as fine as it can for a mobile non-mmx pentium at least.
this is the exact compiler i use [dos_build.zip](https://github.com/Sakitoshi/mbf_sigil/files/13694190/dos_build.zip) allegro is also inside along with a bat file that sets the necessary enviroment variables.
the only edit i'll ask for before merging is change the make optimization cflags to this:
```CFLAGS_RELEASE = -O3 -ffast-math -fomit-frame-pointer  -fstrength-reduce -m486 -fexpensive-optimizations```
seems that -DRANGECHECK causes problems with the frame rate counter and crashes mbf. this is my bad as i was using that, don't know how i didn't noticed before.

dgondos committed:
review comments

dgondos committed:
fix allegro build config

dgondos committed:
fix sigil2 episode text offset

dgondos commented:
Thanks @Sakitoshi!

I made some updates based on your comment as well as [crvs' comments](https://www.vogons.org/viewtopic.php?p=1219297#p1219297) from Vogons:

- Updated makefile with new cflags - tested it with the FPS counter and it works indeed now
- Correctly supporting the case of multiple pwads added with a single `-file` parameter
- Extended the par times array and hardcoded 0 for Sigil II
- G_InitNew now caps the max episode number based on the `modifiedgame` value instead of 6.

Additionally I also fixed a couple of other things:

- The allegro build config needed dpmi as well, updated
- The "Sigil" and "Sigil II" texts in the episode selection screen are using custom graphics and they actually have a larger height than the standard episode select text line height. Sigil II actually even larger than Sigil in fact. This meant that by default the 2px margin between the episodes looked pretty awkward under Sigil (which takes up more vertical space), and I assume when we get Sigil III it'll look even more awkward under Sigil II:

![image](https://github.com/Sakitoshi/mbf_sigil/assets/11691500/6c4913aa-7405-4fe4-ba89-e994855ba810)

To fix that, I changed the menu graphics drawing code so that for the episode selection screen it computes the line height dynamically based on the menu item patch's actual drawn height (instead of assuming uniform 16px). This fixed the spacing between Sigil & Sigil II and it should also work without any changes once Sigil III is added:

![image](https://github.com/Sakitoshi/mbf_sigil/assets/11691500/d25b1851-addf-4ae4-b038-13bf56c8d5bf)

dgondos committed:
update screenshot
